### PR TITLE
Drop hash join reorder as it is currently only partially implemented (and buggy)

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java [hunk 1]
```diff
@@ -43,6 +48,14 @@
 @ESIntegTestCase.ClusterScope(minNumDataNodes = 2)
 public class JoinIntegrationTest extends SQLTransportIntegrationTest {
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        return Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal))
+            .put("stats.service.interval", "2s")
```
Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: Why so big interval here? you can do it even ms I think

Comment (andreidan) [reply sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: The cluster is not ready and it just spams the logs with errors

Comment (mfussenegger) [reply sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: Would it be an option to just get the instance from the cluster state and fill in the stats manually within the tests?

Comment (andreidan) [reply sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: That's a great idea! Giving it a try now

File: sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java [hunk 2]
```diff
@@ -816,4 +819,34 @@ public void testInnerEquiJoinUsingHashJoin() {
                "4| 4\n" +
                "4| 4\n"));
     }
+
+    @Test
```
Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: Shouldn't we use @UseHashJoins(1) to always go to this branch?

File: sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java [hunk 3]
```diff
@@ -816,4 +819,58 @@ public void testInnerEquiJoinUsingHashJoin() {
...
+               "3| 3\n" +
+               "3| 3\n" +
+               "3| 3\n" +
+               "4| 4\n" +
+               "4| 4\n"));
+    }
+
+    @Test
+    // some implementations will apply the branch reordering optimisation, whilst others might not
+    // either way, all join implementations should yield the same results
```
Comment (seut) [inline sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: pls use multiline comment style and also above the annotation

File: sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java [hunk 4]
```diff
@@ -816,4 +819,63 @@ public void testInnerEquiJoinUsingHashJoin() {
...
+        Iterable<TableStats> tableStatsOnAllNodes = internalCluster().getInstances(TableStats.class);
+        for (TableStats tableStats : tableStatsOnAllNodes) {
+            ObjectObjectHashMap<TableIdent, TableStats.Stats> newStats = new ObjectObjectHashMap<>();
+            newStats.put(new TableIdent(sqlExecutor.getDefaultSchema(), "t1"), new TableStats.Stats(2L, 8L));
+            newStats.put(new TableIdent(sqlExecutor.getDefaultSchema(), "t2"), new TableStats.Stats(3L, 12L));
+            newStats.put(new TableIdent(sqlExecutor.getDefaultSchema(), "t3"), new TableStats.Stats(10L, 40L));
+            tableStats.updateTableStats(newStats);
+        }
+
+        execute("select a, x, y from t1 join t2 on t1.a = t2.x join t3 on t3.y = t2.x where abs(t1.a) < abs(t2.x) + 1 " +
```
Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: Is there a reason for using a function (`abs()`) for the filtering?

Comment (andreidan) [reply sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: To always pass the '!query.symbolType().isValueSymbol(),' condition for filtering (see where NestedLoopJoin is created)

Comment (matriv) [reply sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java]: isValueSymbol means the whole query is a Literal, eg: WHERE 1=1

**Pull request merged**